File: 822-unique-morse-code-words/unique-morse-code-words.kt
	class Solution {
		fun uniqueMorseRepresentations(words: Array<String>): Int {
			val morse = arrayOf(".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--..")
			val set = mutableSetOf<String>()
			for (word in words) {
				val sb = StringBuilder()
				for (c in word) {
					sb.append(morse[c - 'a'])
				}
				set.add(sb.toString())
			}
			return set.size

		}
	}

File: 822-unique-morse-code-words/unique-morse-code-words.swift
class Solution {
    func uniqueMorseRepresentations(_ words: [String]) -> Int {
        let dotLetters = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]

        var arr = [String]()
        for i in words {
            let ascii = i.map { $0.asciiValue! - 97}
            var result = ""
            for j in ascii {
                result += dotLetters[Int(j)]
            }
            arr.append(result)
        }

        return Set(arr).count
    }
}